presto-session-property-managers/src/main/java/com/facebook/presto/session/db/DbSessionPropertyManager.java

```diff
@@ -0,0 +1,77 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.session.db;
+
+import com.facebook.presto.session.SessionMatchSpec;
+import com.facebook.presto.spi.session.SessionConfigurationContext;
+import com.facebook.presto.spi.session.SessionPropertyConfigurationManager;
+import com.google.common.collect.ImmutableMap;
+
+import javax.inject.Inject;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A {@link SessionPropertyConfigurationManager} implementation that connects to a database for fetching information
+ * about session property overrides given {@link SessionConfigurationContext}.
+ */
+public class DbSessionPropertyManager
+        implements SessionPropertyConfigurationManager
+{
+    private final DbSpecsProvider specsProvider;
+
+    @Inject
+    public DbSessionPropertyManager(DbSpecsProvider specsProvider)
+    {
+        this.specsProvider = requireNonNull(specsProvider, "specsProvider is null");
+    }
+
+    @Override
+    public SystemSessionPropertyConfiguration getSystemSessionProperties(SessionConfigurationContext context)
+    {
+        List<SessionMatchSpec> sessionMatchSpecs = specsProvider.get();
+
+        // later properties override earlier properties
+        Map<String, String> defaultProperties = new HashMap<>();
+        Set<String> overridePropertyNames = new HashSet<String>();
+        for (SessionMatchSpec sessionMatchSpec : sessionMatchSpecs) {
+            Map<String, String> newProperties = sessionMatchSpec.match(context);
+            defaultProperties.putAll(newProperties);
+            if (sessionMatchSpec.getOverrideSessionProperties().orElse(false)) {
+                overridePropertyNames.addAll(newProperties.keySet());
+            }
+        }
+
+        // Once a property has been overridden it stays that way and the value is updated by any rule
+        Map<String, String> overrideProperties = new HashMap<>();
+        for (String propertyName : overridePropertyNames) {
+            overrideProperties.put(propertyName, defaultProperties.get(propertyName));
+        }
+
+        return new SystemSessionPropertyConfiguration(ImmutableMap.copyOf(defaultProperties), ImmutableMap.copyOf(overrideProperties));
+    }
+
+    @Override
+    public Map<String, Map<String, String>> getCatalogSessionProperties(SessionConfigurationContext context)
+    {
+        // NOT IMPLEMENTED YET
+        return ImmutableMap.of();
+    }
+}
```

Comment:
We shouldn't override these. Based on what I understand, we can just override the abstract method: `getSessionMatchSpecs` in all implementations of `AbstractSessionPropertyManager`